### PR TITLE
ADD MQTT retain feature

### PR DESCRIPTION
**Pull request by fgalan**
Issue #4388

- [x] CRUD of the new field
- [x] Effectively use the new field to publish MQTT notifications
- [x] .test covering the CRUD functionality
- [x] doc
- [x] CNR
- [x] Align existing .test with the new GET subscription(s) output - 19f3d2ccfd6a0533f10290f62c93e4d14163f69d + 9789d5bbf
- [x] Effectively test the new functionality (.test ~or manual testing?~) - 19f3d2ccfd6a0533f10290f62c93e4d14163f69d

### CODE REVIEW

File: doc/manuals/admin/database_model.md [hunk 1]
```diff
@@ -283,8 +283,9 @@ Fields:
     is updated each time a notification is sent, to avoid violating throttling.
 -   **throttling**: minimum interval between notifications. 0 or -1 means no throttling.
 -   **reference**: the URL for notifications, either HTTP or MQTT
--   **mqttTopic**: MQTT topic (only in MQTT notifications)
--   **mqttQoS**: MQTT QoS value (only in MQTT notifications)
+-   **topic**: MQTT topic (only in MQTT notifications)
+-   **qos**: MQTT QoS value (only in MQTT notifications)
```
Comment (fgalan) [inline doc/manuals/admin/database_model.md:287]: Taking into account this at src/lib/mongoBackend/dbConstants.h

```
#define CSUB_MQTTTOPIC               "topic"
#define CSUB_MQTTQOS                 "qos"
```

It seems `mqttQoS` and `mqttTopic` was a typo.

NTC (informative)

**Review by mapedraza — APPROVED** [review]
LGTM

**Comment by mapedraza** [comment]
Seems those tests need to be updated:
https://github.com/telefonicaid/fiware-orion/blob/2f74717b415fbebaf163dad21544711995d17697/test/functionalTest/cases/3914_mqtt_notifications_auth/mqtt_notifications_auth.test
https://github.com/telefonicaid/fiware-orion/blob/2f74717b415fbebaf163dad21544711995d17697/test/functionalTest/cases/3914_mqtt_notifications_auth/mqtt_custom_notifications_auth.test#L4

**Comment by fgalan** [comment]
@fisuda this PR does some modifications to the English documentation. It would be great if you could sync the Japanese translation in a PR (using either feature/4388_mqtt_retain or master as base branch, as you prefer), please. Thanks in advance!

**Comment by fisuda** [comment]
I sent the PR #4413 to the feature/4388_mqtt_retain branch.
Thanks.

**Pull request merged**